src/frontend/hooks/server/media.ts

```diff
@@ -1,67 +1,76 @@
+import {useAttachmentUrl, useCreateBlob} from '@comapeo/core-react';
+import type {BlobId, BlobVariant} from '@comapeo/core/dist/types';
+import type {Observation} from '@comapeo/schema';
+import {useMutation} from '@tanstack/react-query';
 import {URL} from 'react-native-url-polyfill';
-import {useCreateBlob, useAttachmentUrl} from '@comapeo/core-react';
-import {Observation} from '@comapeo/schema';
 import {useActiveProject} from '../../contexts/ActiveProjectContext';
-import {
-  isProcessedDraftPhoto,
-  isUnsavedAudio,
-} from '../../lib/attachmentTypeChecks';
-import {ProcessedDraftPhoto} from '../../contexts/PhotoPromiseContext/types';
-import {UnsavedAudio} from '../../sharedTypes/audio';
-import {BlobId, BlobVariant} from '@comapeo/core/dist/types';
-import {useMutation} from '@tanstack/react-query';
+import type {ProcessedDraftPhoto} from '../../contexts/PhotoPromiseContext/types';
+import type {Attachment} from '../../sharedTypes';
+import type {UnsavedAudio} from '../../sharedTypes/audio';
 
-interface Attachment {
-  driveDiscoveryId: string;
-  name: string;
-  type: 'photo' | 'audio' | 'video';
-  hash: string;
-}
+export function useCreatePhotoAttachment({projectId}: {projectId: string}) {
+  const {mutateAsync: createBlobAsync} = useCreateBlob({projectId});
 
-type AttachmentParam = ProcessedDraftPhoto | UnsavedAudio;
+  return useMutation({
+    mutationFn: async (photo: ProcessedDraftPhoto): Promise<Attachment> => {
+      const blob = await createBlobAsync({
+        original: new URL(photo.originalUri).pathname,
+        preview: photo.previewUri
+          ? new URL(photo.previewUri).pathname
+          : undefined,
+        thumbnail: photo.thumbnailUri
+          ? new URL(photo.thumbnailUri).pathname
+          : undefined,
+        // TODO: DraftPhoto type should probably carry MIME type info that feeds this
+        // although backend currently only uses first part of path
+        metadata: {
+          mimeType: 'image/jpeg',
+          location: photo.mediaMetadata.location,
+          timestamp: photo.mediaMetadata.timestamp,
+        },
+      });
 
-export function useCreateAttachment() {
-  const {projectId} = useActiveProject();
-  const {mutateAsync: createBlobAsync} = useCreateBlob({projectId});
+      const baseAttachment = {
+        driveDiscoveryId: blob.driveId,
+        hash: blob.hash,
+        name: blob.name,
+        type: blob.type,
+      };
+
+      // Should not happen but just in case...
+      if (blob.type !== 'photo') {
+        return baseAttachment;
+      }
```

Comment:
wondering if I should handle this differently. In most cases, this won't happen but to be respectful of TypeScript types, I have this for now.